### PR TITLE
accept new dockercfg format

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/5517

This makes it possible to easily create a docker cfg secret in the new format.

### CODE REVIEW

**Comment by deads2k** [comment]
[test][extended:cmd]

**Comment by deads2k** [comment]
I'll tidy up to use templates tomorrow

**Comment by liggitt** [comment]
bad ami, fix in progress

**Comment by deads2k** [comment]
tests updated to use templates.  I'll chase the ugly jsonpath messages separately.

**Comment by deads2k** [comment]
opened https://github.com/kubernetes/kubernetes/issues/17016

**Comment by deads2k** [comment]
@fabianofranz can you take a look at this one too?

**Comment by deads2k** [comment]
unit test timeout for crypto re[test]

**Comment by deads2k** [comment]
[test]

**Comment by fabianofranz** [comment]
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1282426, correct?

File: test/extended/cmd.sh [hunk 1]
```diff
@@ -77,3 +88,58 @@ oc exec test-pod env | grep podname_composed=test-pod_composed
...
+
+DOCKER_CONFIG_JSON=${HOME}/.docker/config.json
+oc new-project dc-ns
+oc delete all --all
+oc delete secrets --all
+oc secrets new image-ns-pull .dockerconfigjson=${DOCKER_CONFIG_JSON}
+oc secrets new-dockercfg image-ns-pull-old --docker-email=fake@example.org --docker-username=imagensbuilder --docker-server=${DOCKER_REGISTRY} --docker-password=${IMAGE_NS_TOKEN}
+
+oc process -f test/extended/fixtures/image-pull-secrets/pod-with-no-pull-secret.yaml --value=DOCKER_REGISTRY=${DOCKER_REGISTRY} | oc create -f - 
+wait_for_command "oc describe pod/no-pull-pod | grep 'Back-off pulling image'" 30*TIME_SEC
```
Comment (fabianofranz) [inline test/extended/cmd.sh]: Why the sleep?

Comment (fabianofranz) [inline test/extended/cmd.sh]: Nevermind, I just saw the TODO comment.

**Comment by fabianofranz** [comment]
LGTM

**Comment by deads2k** [comment]
> Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1282426, correct?

Yes

**Comment by deads2k** [comment]
[merge]

**Comment by deads2k** [comment]
dockerhub re[merge]

**Pull request merged**